app/api/github/setup/route.ts

```diff
@@ -0,0 +1,206 @@
+  } catch {
+    return null;
+  }
+};
+
+const createOrUpdateFile = async (
+  octokit: Awaited<ReturnType<typeof getOctokit>>,
+  owner: string,
+  repo: string,
+  path: string,
+  content: string,
+  message: string,
+  branch: string,
+) => {
+  try {
+    const sha = await getFileSHA(octokit, owner, repo, path, branch);
+    const params = {
+      owner,
+      repo,
+      path,
+      message,
+      content: Buffer.from(content).toString("base64"),
+      branch,
+      ...(sha ? { sha } : {}),
+    };
+
+    await octokit.rest.repos.createOrUpdateFileContents(params);
+    return { success: true };
+  } catch (error: any) {
+    if (error.status === 409) {
+      throw new Error(`Conflict while updating ${path}. Please try again.`);
+    }
+    if (error.status === 422) {
+      throw new Error(
+        `Validation failed for ${path}. Please check file content and permissions.`,
+      );
+    }
+    if (error.status === 404) {
+      throw new Error(`Path ${path} not found or insufficient permissions.`);
+    }
+    throw error;
+  }
+};
+
+export const POST = async (request: NextRequest) => {
+  const { userId } = auth();
+
+  if (!userId) {
+    return NextResponse.redirect(new URL("/login", request.url));
+  }
+
+  try {
+    const octokit = await getOctokit();
+    const body = await request.json();
+
+    const validatedData = SetupRequestSchema.parse(body) satisfies SetupRequest;
+    const { repoFullName, testPattern, triggers, defaultBranch } =
+      validatedData;
+    const [owner, repo] = repoFullName.split("/");
+
+    try {
+      await octokit.rest.repos.get({
+        owner,
+        repo,
+      });
+    } catch (error) {
+      console.error("Error accessing repository:", error);
+      return NextResponse.json(
+        {
+          error:
+            "Repository not found or not accessible. Please make sure the repository exists and you have the right permissions.",
+        },
+        { status: 404 },
+      );
+    }
+
+    await createOrUpdateRepositoryConfig({
+      owner,
+      repo,
+    });
+
+    const configContent = `import type { ShortestConfig } from "@antiwork/shortest";
+
+export default {
+  headless: false,
+  baseUrl: "http://localhost:3000",
+  testPattern: '${testPattern}',
+  ai: {
+    provider: "anthropic",
+    apiKey: process.env.ANTHROPIC_API_KEY
+  }
+} satisfies ShortestConfig;`;
+
+    const workflowContent = `name: Shortest E2E Tests
+
+on:
+${triggers.onPush ? `  push:\n    branches:\n      - ${defaultBranch}\n` : ""}
+${triggers.onPullRequest ? "  pull_request:\n" : ""}
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    env:
+      ANTHROPIC_API_KEY: \${{ secrets.ANTHROPIC_API_KEY }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: '22'
+      - name: Install dependencies
+        run: npm install
+      - name: Run Shortest tests
+        run: npx shortest run`;
```

Comment:
Shortest needs a web server to run. It will try to hit `http://localhost:3000/` that is not available.

Reply:
Based on [the assumptions](https://github.com/antiwork/shortest/issues/384#issuecomment-2712883594) on the associated issue,

Then we should ask the user for the start script of the app & the port (so we can use those on the workflow & config) - we could assume the start script, but probably not the port.

Reply:
> Shortest needs a web server to run. It will try to hit `http://localhost:3000/` that is not available.

Based on [the assumptions](https://github.com/antiwork/shortest/issues/384#issuecomment-2712883594) on the associated issue,

Then we could ask the user for the start script of the app & the port (so we can use those on the workflow & config) - but that would require the Github env to have all necessary variables to run the app. I believe if someone wants to do e2e tests, that's something the should expect though.

The other option would be to simply use a hosted app URL and have this workflow only trigger manually? (so the user can control when the deployment is up and ready)

Reply:
Assuming that MVP only supports Vercel, we can generate a workflow similar to this
https://github.com/antiwork/shortest/blob/main/.github/workflows/web-shortest.yml

However, we'd also need to figure out what to do with the package manager (we could have support for the most popular ones, and generate workflow with the install scripts based on the configured package manager?)

Reply:
The same with other setup scripts (i.e. the drizzle scripts in the web-shortest.yml)